### ISSUE TITLE
[ISSUE] Incorrect internal constraints on VGSCardTextField

### ISSUE THREAD

**Issue by DagAgren** [verygoodsecurity/vgs-collect-ios#162]
Constraining the height of a VGSCardTextField causes an autolayout conflict. If the constraint is removed, the field will be much too large, indicating that there are incorrect internal constraints forcing the size to something unexpected.

This is what the view looks like without a height constraint: 
<img width="487" alt="Screenshot 2020-06-05 at 17 09 04" src="https://user-images.githubusercontent.com/11536828/83886643-087f3c80-a750-11ea-8a58-582c27b03f4f.png">

This is the error output we get when we try to constrain the height to 44:

```
    (
        "<NSLayoutConstraint:0x600001a3c460 V:|-(8)-[UIStackView:0x7f8b2eedf900]   (active, names: '|':VGSCollectSDK.VGSCardTextField:0x7f8b2eee9490 )>",
        "<NSLayoutConstraint:0x600001a3c0a0 V:[UIStackView:0x7f8b2eedf900]-(8)-|   (active, names: '|':VGSCollectSDK.VGSCardTextField:0x7f8b2eee9490 )>",
        "<NSLayoutConstraint:0x600001990aa0 VGSCollectSDK.VGSCardTextField:0x7f8b2eee9490.height == 45   (active)>",
        "<NSLayoutConstraint:0x6000019dcc30 'heightConstraint' UIImageView:0x7f8b2eede7e0.height == 45   (active)>",
        "<NSLayoutConstraint:0x600001a070c0 'UISV-alignment' VGSCollectSDK.MaskedTextField:0x7f8b2fa52000.bottom == UIImageView:0x7f8b2eede7e0.bottom   (active)>",
        "<NSLayoutConstraint:0x600001a13d40 'UISV-alignment' VGSCollectSDK.MaskedTextField:0x7f8b2fa52000.top == UIImageView:0x7f8b2eede7e0.top   (active)>",
        "<NSLayoutConstraint:0x6000019dc5a0 'UISV-canvas-connection' UIStackView:0x7f8b2eedf900.top == VGSCollectSDK.MaskedTextField:0x7f8b2fa52000.top   (active)>",
        "<NSLayoutConstraint:0x6000019dc550 'UISV-canvas-connection' V:[VGSCollectSDK.MaskedTextField:0x7f8b2fa52000]-(0)-|   (active, names: '|':UIStackView:0x7f8b2eedf900 )>"
    )
```

**Comment by dmytrokhl** [comment]
@DagAgren thanks for detailed example. I assume this is card icon height constraint priority issue. We will try to fix it in next release.

**Comment by dmytrokhl** [comment]
@DagAgren as a temporary solution you can try to set cardIconSize with `height` = `textfield height` - `topPadding` - `bottomPadding`. 

Example:
 `cardNumber.cardIconSize = CGSize(width: 45, height: 32)`

**Comment by DagAgren** [comment]
It might make more sense to constrain the icon to centerX and not to top or bottom at all?

**Comment by dmytrokhl** [comment]
Card icon has only width & height constraints and is arranged in stackview, so it helps to make the icon size more dynamic and don't allows text to go under the image. I'm not sure we can constrain the icon to centerX in stackview, maybe we can just limit max icon height by stackview height.

**Comment by dmytrokhl** [comment]
The fix released in [v1.5.4](https://github.com/verygoodsecurity/vgs-collect-ios/releases/tag/1.5.4).
